pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -647,9 +647,19 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 		} else {
 			logger.Warningf("failed to list mon deployments. attempting to continue. %v", err)
 		}
-	} else if len(deployments.Items) < len(mons) {
+	}
+
+	readyReplicas := 0
+	// Ensuring the mon deployments should be ready
+	for _, deploy := range deployments.Items {
+		readyReplicas = int(deploy.Status.AvailableReplicas) + readyReplicas
```

Comment:
Something that has been tugging at my brain, should we do a little more here and do the following? Just in case the user (or we in the future) make it possible to have a mon deployment with multiple replicas, we don't want to over-count.

```go
if deploy.Status.AvailableReplicas > 0 {
    readyReplicas += 1
}
```

Reply:
Agreed that's more correct even though I'm not sure what behavior you would see in the first place if someone tried to start multiple replicas of the same mon.

Reply:
yup addressed